sycl/source/handler.cpp

```diff
@@ -57,6 +57,10 @@ handler::getOrInsertHandlerKernelBundle(bool Insert) const {
   if (!KernelBundleImpPtr && Insert) {
     KernelBundleImpPtr = detail::getSyclObjImpl(
         get_kernel_bundle<bundle_state::input>(MQueue->get_context()));
+    if (KernelBundleImpPtr->empty()) {
```

Comment:
NIT. It would be nice to have a comment explaining this logic.

Reply:
I'll submit some comments as a separate PR.